lighthouse-core/test/lib/page-functions-test.js

```diff
@@ -177,16 +177,16 @@ describe('Page Functions', () => {
       assert.equal(pageFunctions.getNodeLabel(el), Array(78).fill('a').join('') + '💡…');
     });
 
-    it('Uses tag name for html tags', () => {
+    it('Returns null if nodeLabel is not usable for html tags', () => {
```

Comment:
Can you add a test case to `getNodeDetails` that covers the fallack?

Reply:
Yes! That's where my mind skipped, thank you, will commit change shortly :)

Reply:
Well atleast its using the `selector` as fallback like intended 😄 , once the test is added to cover fallback I think the smoke test should pass.

✘ difference at aria-allowed-attr audit.details.items[0].node.nodeLabel
              expected: "div"
                 found: "body > section > div#aria-allowed-attr"

Reply:
@connorjclark When I try to use 
```
const el = dom.createElement("html");
assert.equal(pageFunctions.getNodeDetails(el), { ... nodeLabel: 'html' }` 
```
its saying I should be using jsdom as `window isn't defined`. Am I missing some small detail to write this test for `getNodeDetails()`?

Reply:
you can add a `global.window = {}` / `global.window = undefined` to make that work–the first few lines of `getNodeDetails` (which i'm just realizing we have no unit tests for... on smoke tests) expected there to be a `window` variable in the global scope.

Reply:
yeah that did the trick, thanks! I think I expected there to be a `window` variable in the global scope already too and was just stumped because there weren't any other unit tests that tested `getNodeDetails` and needed `window`. 

>  (...which i'm just realizing we have no unit tests for... only smoke tests)

I noticed that while looking through `page-functions-test.js` as well, I'm happy to help write a few more unit tests to test the other fields in `details` like `devtoolsNodePath`,` selector`, `boundingRect` etc similar to how were testing `nodeLabel`.

Reply:
This is a simple test to make the `selector` value is being used as the fallback for `nodeLabel` in `details`. We could pluck each property from `details` using destructuring assignment to write a few unit tests for the other properties besides `nodeLabel`. Might need to be robust for other properties in `details` but I thought this was a start for potentially writing tests for the others down the line.

```js
it('Returns selector as fallback if nodeLabel equals html tag name', () => {
   // define a global `window` object for the first
   // few lines in `getNodeDetails`
   global.window = {};
   // use `html` element to ensure getNodeLabel returns null
   const el = dom.createElement('html');
   const {nodeLabel} = pageFunctions.getNodeDetails(el);
   assert.equal(nodeLabel, 'html');
});
```

Reply:
@connorjclark I think the above unit test could be a little more sturdy but let me know your thoughts.

Reply:
The selector is the fallback now, not the tag name.

If you make a nested element, and give both of them a class and id, that should suffice for testing purposes.

Reply:
> The selector is the fallback now, not the tag name.

Yeah I understood that `nodeLabel: getNodeLabel(..) || selector` should be using `selector` as fallback when getNodeLabel is null. I think my tests didn't fully illustrate that though, will update with the suggested improvements.

(ok thanks for the heads up, I will make sure to avoid sharing snippets in comments in the future)